docs/ui.md

```diff
@@ -27,10 +27,83 @@ These elements are defined in the manifest.xml file:
 </connector-plugin>
 ```
 
+## Vendor Attributes (Custom Fields)
+Vendors can add customized attributes to their connector plugin by using the vendor attributes.
+
+These fields have a custom label and can be used for attributes in the connection strings that are not available in the attribute list. You can currently add 3 custom fields in your connector plugin.
+
+To add a custom vendor-attribute, you will need to modify your connection-dialog.tcd, connectionResolver.tdr and connectionBuilder.js.
+**For JDBC based plugins, you will need to modify connectionProperties.js instead of connectionBuilder.js**
+
+connection-resolver.tdr
+
+```xml
+    ...
+      <required-attributes>
+      <attribute-list>
+        ...
+        <attr> vendor1 </attr> 
+        <attr> vendor2 </attr> 
+        <attr> vendor3 </attr> 
+
+      </attribute-list>
+      </required-attributes>
+    ...
+```
+
+connection-dialog.tcd
+
+```xml
+ <connector-plugin class='postgres_jdbc' superclass='jdbc' plugin-version='0.0.0' name='PostgreSQL JDBC' version='18.1'>
+          <connection-config>
+            ...
+            <vendor1-prompt value="Log Level: "/>
+            <vendor2-prompt value="Protocol Version: "/>
+            <vendor3-prompt value="Char Set: "/>
+
+        </connection-config>
+      </connection-dialog>
+```
+
+connectionBuilder.js (Non-JDBC)
```

Comment:
Non-JDBC is a little confusing, its ODBC.  We can do this later, but really we have ODBC and JDBC based connectors only right now.  We might consider structuring the docs around ODBC, JDBC, future, etc eventually.

Reply:
Changed this to ODBC